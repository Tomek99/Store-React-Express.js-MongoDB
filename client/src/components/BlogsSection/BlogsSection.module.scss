.BlogsSection {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    color: #fff;

    .blogs {
        max-width: 1366px;
        width: 100%;
        flex-wrap: wrap;
        display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: space-between;
    }
}

@media screen and (max-width: 900px) {
    .blogs {
        max-width: 600px !important;
    }
}

@media screen and (max-width: 500px) {
    .blogs {
        max-width: 300px !important;
    }
}
