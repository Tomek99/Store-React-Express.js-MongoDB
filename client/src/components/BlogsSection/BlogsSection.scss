.blogsSection {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: #fff;

  .blogsCarousel {
    max-width: 1366px;
    width: 100%;
    overflow: hidden;
  }
}

@media (max-width: 1366px) {
  .blogsCarousel {
    max-width: 1280px !important;
  }
}

@media (max-width: 1280px) {
  .blogsCarousel {
    max-width: 1024px !important;
  }
}

@media (max-width: 1024px) {
  .blogsCarousel {
    max-width: 900px !important;
  }
}

@media (max-width: 900px) {
  .blogsCarousel {
    max-width: 730px !important;
  }
}

@media (max-width: 730px) {
  .blogsCarousel {
    max-width: 580px !important;
  }
}

@media (max-width: 580px) {
  .blogsCarousel {
    max-width: 450px !important;
  }
}
@media (max-width: 450px) {
  .blogsCarousel {
    max-width: 320px !important;
  }
}
