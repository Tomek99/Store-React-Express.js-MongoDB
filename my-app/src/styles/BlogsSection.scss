div.BlogsSectionn {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    color: #fff;

    div.blogs {
        max-width: 900px;
        display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: center;
        gap: 10px;

        div.singleBlog {
            flex-basis: 32.4%;
            display: flex;
            flex-direction: column;
            align-items: flex-start;
            gap: 10px;
            border: var(--border);
            overflow: hidden;
           
            div.zoomImage{
                overflow: hidden;
                img {
                    width: 100%;
              
                }
            }

            div.textSection {
                display: flex;
                flex-direction: column;
                align-items: flex-start;
                gap:10px;
                margin: 10px;
                
                h2 {
                    font-size: 2rem;
                    cursor: pointer;
                }

                h2:hover {
                    color: #ccc;
                }

                h4 {
                    font-size: 1.6rem;
                    font-weight: 400;
                    color:var(--main-color)
                }
                
                p {
                    font-size: 1.3rem;
                    color: #ccc;
                }

                button {
                    background-color: var(--main-color);
                    color: #fff;
                    font-size: 1.6rem;
                    padding: 6px 17.5px 6px 17.5px;
                    cursor: pointer;
                }
                button:hover{
                    letter-spacing: .1rem;
                }
            }
        }

        div.singleBlog:hover img {
            transform: scale(1.1);
        }
    }

}

@media screen and (max-width: 800px) {
    div.blogs {
       max-width: 600px !important;
       flex-direction: column !important;

    }
}

