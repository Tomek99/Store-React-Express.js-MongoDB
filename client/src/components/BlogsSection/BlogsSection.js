import React from "react";
import styles from "./BlogsSection.module.scss";
import HeaderSection from "../HeaderSection/HeaderSection";
import LatestBlog from "./LatestBlog/LatestBlog";
import Blog from "../../data/blog.json";
import PropTypes from "prop-types";

function BlogsSection() {
  return (
    <div className={styles.BlogsSection} id="blogsSection">
      <HeaderSection firstWord="our" secondWord="news" />
      <div className={styles.blogs}>
        {Blog.slice(0, 3).map((item, index) => (
          <LatestBlog key={index} item={item} />
        ))}
      </div>
    </div>
  );
}
BlogsSection.propTypes = {
  isTrue: PropTypes.bool,
};

export default BlogsSection;
