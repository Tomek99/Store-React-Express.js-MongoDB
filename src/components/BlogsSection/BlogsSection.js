import React from "react";
import styles from "./BlogsSection.module.scss";
import HeaderSection from "../HeaderSection/HeaderSection";
import LatestBlog from "../LatestBlog/LatestBlog";
import Blog from "../../data/blog.json";

function BlogsSection(props) {
  return (
    <div className={styles.BlogsSection} id="blogsSection">
      <HeaderSection firstWord="our" secondWord="blogs" />
      <div className={styles.blogs}>
        {Blog.map((item) => (
          <LatestBlog
            url={item.imageUrl}
            id={item.id}
            title={item.title}
            text={item.text.slice(0, 55)}
            addedBy={item.addedBy}
            handleBlog={props.handleBlog}
          />
        ))}
      </div>
    </div>
  );
}

export default BlogsSection;
