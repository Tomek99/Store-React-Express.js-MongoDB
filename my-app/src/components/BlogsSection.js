import React from 'react'
import '../styles/BlogsSection.scss'
import Header from './Header'
import LatestBlog from './LatestBlog'
import Blog from '../data/blog.json';

function BlogsSection() {
    return (
        <div className='BlogsSectionn' id='blogsSection'>
            <Header firstWord="our" secondWord="blogs" />
            <div className='blogs'>
                {Blog.map((item) => <LatestBlog url={item.imageUrl} id={item.id} title={item.title} text={item.text} addedBy={item.addedBy} />)}
            </div>
        </div>
    )
}

export default BlogsSection