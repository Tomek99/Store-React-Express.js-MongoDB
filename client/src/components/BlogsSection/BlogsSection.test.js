import { render, screen } from "@testing-library/react";
import BlogsSection from "./BlogsSection";

test("should render BlogsSection component", () => {
  const { container } = render(<BlogsSection />);

  expect(container).toMatchSnapshot();
});
