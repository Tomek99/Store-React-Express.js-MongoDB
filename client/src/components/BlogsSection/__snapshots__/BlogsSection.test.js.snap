// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BlogsSection component 1`] = `
<div>
  <div
    class="BlogsSection"
    id="blogsSection"
  >
    <header
      class="headerSection"
    >
      <span>
        our
      </span>
       
      <span>
        news
      </span>
    </header>
    <div
      class="blogs"
    >
      <div
        class="singleBlog"
      >
        <div
          class="zoomImage"
        >
          <img
            alt="Tasty And Refreshing Coffee"
            src="images/blog-1.jpeg"
          />
        </div>
        <div
          class="textSection"
        >
          <h2>
            Tasty And Refreshing Coffee
          </h2>
          <h4>
            By Admin / 1st May, 2021
          </h4>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit
          </p>
        </div>
      </div>
      <div
        class="singleBlog"
      >
        <div
          class="zoomImage"
        >
          <img
            alt="Tasty And Refreshing Coffee"
            src="images/blog-2.jpeg"
          />
        </div>
        <div
          class="textSection"
        >
          <h2>
            Tasty And Refreshing Coffee
          </h2>
          <h4>
            By Admin / 2st May, 2021
          </h4>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit
          </p>
        </div>
      </div>
      <div
        class="singleBlog"
      >
        <div
          class="zoomImage"
        >
          <img
            alt="Tasty And Refreshing Coffee"
            src="images/blog-3.jpeg"
          />
        </div>
        <div
          class="textSection"
        >
          <h2>
            Tasty And Refreshing Coffee
          </h2>
          <h4>
            By Admin / 3st May, 2021
          </h4>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
